core/src/main/java/io/substrait/type/proto/ParameterizedProtoConverter.java

```diff
@@ -197,33 +201,54 @@ public ParameterizedType map(ParameterizedType key, ParameterizedType value) {
     @Override
     protected ParameterizedType wrap(final Object o) {
       var bldr = ParameterizedType.newBuilder();
-      return switch (o) {
-        case Type.Boolean t -> bldr.setBool(t).build();
-        case Type.I8 t -> bldr.setI8(t).build();
-        case Type.I16 t -> bldr.setI16(t).build();
-        case Type.I32 t -> bldr.setI32(t).build();
-        case Type.I64 t -> bldr.setI64(t).build();
-        case Type.FP32 t -> bldr.setFp32(t).build();
-        case Type.FP64 t -> bldr.setFp64(t).build();
-        case Type.String t -> bldr.setString(t).build();
-        case Type.Binary t -> bldr.setBinary(t).build();
-        case Type.Timestamp t -> bldr.setTimestamp(t).build();
-        case Type.Date t -> bldr.setDate(t).build();
-        case Type.Time t -> bldr.setTime(t).build();
-        case Type.TimestampTZ t -> bldr.setTimestampTz(t).build();
-        case Type.IntervalYear t -> bldr.setIntervalYear(t).build();
-        case Type.IntervalDay t -> bldr.setIntervalDay(t).build();
-        case ParameterizedType.ParameterizedFixedChar t -> bldr.setFixedChar(t).build();
-        case ParameterizedType.ParameterizedVarChar t -> bldr.setVarchar(t).build();
-        case ParameterizedType.ParameterizedFixedBinary t -> bldr.setFixedBinary(t).build();
-        case ParameterizedType.ParameterizedDecimal t -> bldr.setDecimal(t).build();
-        case ParameterizedType.ParameterizedStruct t -> bldr.setStruct(t).build();
-        case ParameterizedType.ParameterizedList t -> bldr.setList(t).build();
-        case ParameterizedType.ParameterizedMap t -> bldr.setMap(t).build();
-        case Type.UUID t -> bldr.setUuid(t).build();
-        default -> throw new UnsupportedOperationException(
-            "Unable to wrap type of " + o.getClass());
-      };
+      if (o instanceof Type.Boolean t) {
```

Comment:
nit: this code is duplicated 3 times. It seems to be a good candidate for refactoring.